clocking-server/src/tcp/certs.rs

```diff
@@ -15,16 +17,80 @@ pub struct SingleCerts {
     pub keys: PrivateKeyDer<'static>,
 }
 
-impl SingleCerts {
-    pub fn new(certs_path: &Path, keys_path: &Path) -> io::Result<Self> {
-        Ok(Self {
-            certs: certs(&mut BufReader::new(File::open(certs_path)?))
+fn load_from_env() -> io::Result<Option<SingleCerts>> {
+    let Ok(certs_env) = env::var("SINGLECERTS_CERT_PEM") else {
+        return Ok(None);
+    };
+    let Ok(keys_env) = env::var("SINGLECERTS_KEY_PEM") else {
+        return Ok(None);
+    };
+
+    let keys = pkcs8_private_keys(&mut keys_env.as_bytes())
+        .next()
+        .ok_or(io::ErrorKind::InvalidInput)?
+        .map(Into::into)?;
+    Ok(Some(SingleCerts {
+        certs: certs(&mut certs_env.as_bytes())
+            .collect::<io::Result<Vec<CertificateDer<'static>>>>()?,
+        keys,
+    }))
+}
+
+fn load_from_env_path() -> io::Result<Option<(SingleCerts, String, String)>> {
+    let Ok(certs_path) = env::var("SINGLECERTS_CERT_PATH") else {
+        error!("Certification file was specified by 'SINGLECERTS_CERT_PATH', but not existing. ");
+        return Ok(None);
+    };
+    let Ok(keys_path) = env::var("SINGLECERTS_KEY_PATH") else {
+        error!("Private key file was specified by 'SINGLECERTS_KEY_PATH', but not existing.");
+        return Ok(None);
+    };
```

Comment:
これ……は違うかにゃ
このエラー文は、「SINGLECERTS_PATH」環境変数が**指定されたけど、指定されたパスに環境変数はないよ** ってエラーだからね (このif文の中はそもそも環境変数が指定されてない時に通過する)
```suggestion
    let Ok(certs_path) = env::var("SINGLECERTS_CERT_PATH") else {
        return Ok(None);
    };
    let Ok(keys_path) = env::var("SINGLECERTS_KEY_PATH") else {
        return Ok(None);
    };
```